src/rust/engine/src/nodes.rs

```diff
@@ -1,8 +1,8 @@
 // Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
 // Licensed under the Apache License, Version 2.0 (see LICENSE).
 
-use std::collections::HashSet;
 use std::error::Error;
+use std::collections::{BTreeMap, HashSet};
```

Comment:
Rather than `BTreeMap`, which sorts its keys, you'll want to use `OrderMap`, which just maintains key ordering. It's already imported in this file.

Reply:
I'm pretty sure that this *should* be a `BTreeMap`; in the remote case, these are required to be sorted, and in the local case, having them be canonical for cache-key purposes is good.

(Incidentally, we should probably error here if a key is present twice :))

Reply:
I think I had been thinking that order was relevant in ENV, but perhaps it is not. So yea, makes sense.